src/couch/src/couch_bt_engine_header.erl

```diff
@@ -285,6 +286,20 @@ upgrade_epochs(#db_header{} = Header) ->
     DedupedEpochs = remove_dup_epochs(NewEpochs),
     Header#db_header{epochs = DedupedEpochs}.
 
+%% get the node name from the start up configuration as node()
+%% can return nonode@nohost if net_kernel crashes.
+init_node_name() ->
+    init_node_name(init:get_arguments()).
+
+init_node_name([]) ->
+    nonode@nohost;
+init_node_name([{name, [Name]} | _Rest]) ->
+    list_to_existing_atom(Name);
+init_node_name([{sname, [Name]} | _Rest]) ->
+    list_to_existing_atom(Name);
+init_node_name([_ | Rest]) ->
+    init_node_name(Rest).
+
```

Comment:
There is some danger here if we don't match exactly the node parsing code from `init` and `net_kernel` we could get into case where our init_node_name doesn't match exactly the output of `node()`.

For instance is someone uses just the name and gets the host automatically from the various resolving paths, this will break, it will also break if someone uses sname. We advise anyone do that, but we did once, and so may break people's setup then.

```
 % erl -noshell -name foo@127.0.0.1 -eval 'io:fwrite("~s~n", [node()]), init:stop().'
foo@127.0.0.1
 % erl -noshell -name foo -eval 'io:fwrite("~s~n", [node()]), init:stop().'
foo@myhostname.local
% erl -noshell -sname foo -eval 'io:fwrite("~s~n", [node()]), init:stop().'
foo@myhostname
```

We don't want to be in the business deal `.local` bits and all that given that it's kind opaque hidden inside net_kernel machinery.